src/components/header/modals/ModalPolkasafe.vue

```diff
@@ -0,0 +1,394 @@
+
+    const substrateAccounts = computed<SubstrateAccount[]>(
+      () => store.getters['general/substrateAccounts']
+    );
+
+    const selectAccount = async (substrateAccount: string): Promise<void> => {
+      await props.disconnectAccount();
+      isClosing.value = true;
+      const animationDuration = 500;
+      await wait(animationDuration);
+      store.commit('general/setCurrentAddress', substrateAccount);
+      localStorage.setItem(LOCAL_STORAGE.SELECTED_WALLET, SupportMultisig.Polkasafe);
+      const multisigObj = {
+        multisigAccount: multisigAccounts.value.find((it) => it.address === substrateAccount),
+        signatory: selectedSignatory.value,
+      } as Multisig;
+      localStorage.setItem(LOCAL_STORAGE.MULTISIG, JSON.stringify(multisigObj));
+      isSelected.value = true;
+      isClosing.value = false;
+      emit('update:is-open', false);
+      window.dispatchEvent(new CustomEvent(LOCAL_STORAGE.SELECTED_WALLET));
+    };
+
+    const selAccount = ref<string>('');
+    const subScan = computed<string>(
+      () => `${providerEndpoints[currentNetworkIdx.value].subscan}/account/`
+    );
+
+    const previousSelIdx = computed<number | null>(() => {
+      if (substrateAccounts.value && props.currentAccount) {
+        const index = substrateAccounts.value.findIndex(
+          (it: SubstrateAccount) => it.address === props.currentAccount
+        );
+        return index;
+      } else {
+        return null;
+      }
+    });
+
+    const copyAddress = (address: string): void => {
+      copy(address);
+      store.dispatch('general/showAlertMsg', {
+        msg: t('toast.copyAddressSuccessfully'),
+        alertType: 'copied',
+      });
+    };
+
+    const windowHeight = ref<number>(window.innerHeight);
+    const onHeightChange = (): void => {
+      const adjustment = width.value > screenSize.sm ? 520 : 390;
+      windowHeight.value = window.innerHeight - adjustment;
+    };
+
+    window.addEventListener('resize', onHeightChange);
+    onHeightChange();
+
+    const displayBalance = (balance: string): number => {
+      return truncate(ethers.utils.formatEther(balance || '0'));
+    };
+
+    const handleGetMultisigAccounts = async (): Promise<void> => {
+      multisigAccounts.value = [];
+      isLoadingPolkasafe.value = true;
+      selAccount.value = '';
+      if (!selectedSignatory.value || currentNetworkChain.value !== astarChain.ASTAR) {
+        return;
+      }
+      try {
+        const extensions = await web3Enable('AstarNetwork/astar-apps');
+        const injector = extensions.find((it) => {
+          return selectedSignatory.value && it.name === selectedSignatory.value.source;
+        }) as any;
+
+        if (!injector) {
+          throw Error('injector not found');
+        }
+        const substratePrefix = 42;
+        const signatory = encodeAddress(selectedSignatory.value.address, substratePrefix);
+
+        const setMultisigAccounts = async (c: Polkasafe): Promise<void> => {
```

Comment:
I have few last comments. Maybe to move this outside `handleGetMultisigAccounts` because it is hard to read the code. Do the same for `handleInitializePolkasafe`.

Reply:
Fixed in [this](https://github.com/AstarNetwork/astar-apps/pull/834/commits/6d8f31f2645b3e46060408ae8ca42e1fa7ba542d) commit.